Source/WebKit/UIProcess/UserContent/WebUserContentControllerProxy.cpp

```diff
@@ -229,8 +229,8 @@ void WebUserContentControllerProxy::removeAllUserScripts(API::ContentWorld& worl
     for (Ref process : m_processes)
         process->send(Messages::WebUserContentController::RemoveAllUserScripts({ world.identifier() }), identifier());
 
-    protectedUserScripts()->removeAllOfTypeMatching<API::UserScript>([&](const auto& userScript) {
-        return &userScript->contentWorld() == &world;
+    protectedUserScripts()->removeAllOfTypeMatching<API::UserScript>([world = Ref { world }](const auto& userScript) {
```

Comment:
We should just add `NOESCAPE` macro on `MatchFunction` in `removeAllOfTypeMatching` instead.